### PR TITLE
elastic search 

### PR DESCRIPTION
**Pull request by slajax**
Overview
===

I really wanted to be able to create managed elastic search resources via the convox cli so I added it. 

It is working right now using a local rack. It set it up so can white list a local IP for `es:*` (super admin) and define a user ARN so that you can sign your requests with it's credentials for `es:ESHttp*` (search, insert etc). 

By default your ECS containers won't even have access to the elastic search resource. You have to sign requests using the ARN key and secret using Signature Version 4 signing as described [here](https://aws.amazon.com/blogs/security/how-to-control-access-to-your-amazon-elasticsearch-service-domain/). I've tested it manually by exec'ing into a container and this is working.

I also haven't added any tests yet, but I'm going to look into that next - I may need some help on that - this is the first time I have ever worked with go and I'm learning as I go _(see what I did there?)_.

Example
===

`❯ convox resources create elasticsearch --domain-name testing --user-arn "arn:aws:iam::123456789101:user/es" --whitelist-ip "8.8.8.8"`

You can also over write the other defaults:

`elasticsearch  [--domain-name] [--user-arn] [--whitelist-ip=127.0.0.1/0] [--version=2.3] [--dedicated-master=true] [--master-instance-type=m3.medium.elasticsearch] [--master-instance-count=3] [--instance-type=m3.medium.elasticsearch] [--instance-count=2] [--iops=0] [--volume-size=20]`

Tasks
===
- [x] create working cloudformation template
- [x] sync variables with cli
- [x] use sane defaults
- [x] default whitelist to 127.0.0.1/0 
- [x] allow overrides
- [x] add to correct places in cli logic
- [ ] links
- [ ] allow fluentd to forward logs to es
- [ ] tests - *handing off to convox team*

### CODE REVIEW

**Comment by slajax** [comment]
@ianmalott please ping me in slack if you get a moment tomorrow.

File: provider/aws/templates/resource/elasticsearch.tmpl [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "Conditions": {
+    },
+    "Parameters": {
+      "DomainName": {
+        "Type": "String",
+        "Description": "Name of elasticsearch domain"
+      },
+      "UserArn": {
+        "Type": "String",
+        "Description": "Name of elasticsearch domain"
```
Comment (ddollar) [inline provider/aws/templates/resource/elasticsearch.tmpl]: Looks like a description accidentally got copy/pasted here

File: provider/aws/templates/resource/elasticsearch.tmpl [hunk 2]
```diff
@@ -0,0 +1,128 @@
+{{ define "resource" }}
+{
+  "AWSTemplateFormatVersion" : "2010-09-09",
+    "Conditions": {
+    },
+    "Parameters": {
```
Comment (ddollar) [inline provider/aws/templates/resource/elasticsearch.tmpl]: Would you please alphabetize these parameters? Makes it easier to manage as things grow.

File: cmd/convox/services.go [hunk 3]
```diff
@@ -37,6 +37,10 @@ func init() {
 			"[--automatic-failover-enabled] [--database=db-name] [--instance-type=cache.t2.micro] [--num-cache-clusters=1] [--private]",
 		},
 		{
+			"elasticsearch",
+			"[--domain-name] [--user-arn] [--whitelist-ip=127.0.0.1/0] [--version=2.3] [--dedicated-master=true] [--master-instance-type=m3.medium.elasticsearch] [--master-instance-count=3] [--instance-type=m3.medium.elasticsearch] [--instance-count=2] [--iops=0] [--volume-size=20]",
```
Comment (ddollar) [inline cmd/convox/services.go]: Should be `[--domain-name=example-domain-here]` etc for parameters that take values.

**Comment by slajax** [comment]
@ddollar here are the changes you requested. I tried creating a new es resource while there is an existing one in a "deleting" state and it looks like I'm getting this: 

`ERROR: ValidationError: Circular dependency between resources: [ElasticsearchDomain]`

I'm going to test again once it's deleted to confirm, but this seems flawed. I should be able to create multiple ES Domains at a given time no?

**Comment by slajax** [comment]
Hmm, nope, looks like it's consistent now. Even if I roll back to the previous commit that was working last night. I'm kind of blocked here I think. Let me know if you have any ideas.

**Comment by slajax** [comment]
This is working well now, it's just missing the tasks as described in issue. The rest of it is a bit over my head, but let me know if I can be of any help with this.

**Comment by ddollar** [comment]
Closing for now since this is quite old but will borrow this cloudformation template for Praxis. Thanks!

**Pull request closed without merging**